modules/tinymce/src/core/test/ts/browser/newline/InsertNewLineTest.ts

```diff
@@ -836,4 +836,15 @@ describe('browser.tinymce.core.newline.InsertNewLineTest', () => {
     TinyAssertions.assertContent(editor, '<blockquote><p>A</p></blockquote><p>&nbsp;</p>');
     TinyAssertions.assertCursor(editor, [ 1, 0, 0, 0 ], 0);
   });
+
+  it('TINY-10560: Press Enter after a `selection.setContent` should create a new line and not throw errors', () => {
+    const editor = hook.editor();
+    editor.setContent('abc');
+    TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 3);
+
+    editor.selection.setContent('<pre><code>hello</code></pre>');
```

Comment:
It's interesting that `editor.insertContent` doesn't fail so maybe the command some how fixes the selection first.

Reply:
it seems that `editor.insertContent` is using `selection.setContent` but with a [bookmark](https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts#L330) and then [move the selection](https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts#L372) to that bookmark